distributed/_ipython_utils.py

```diff
@@ -234,20 +226,17 @@ def _start():
         app.kernel.pre_handler_hook = noop
         app.kernel.post_handler_hook = noop
         app.kernel.start()
-        app.kernel.loop = IOLoop.instance()
         # save self in the IPython namespace as 'worker'
         # inject things into the IPython namespace
         if ns:
             app.kernel.shell.user_ns.update(ns)
         evt.set()
-        zmq_loop.start()
+        # start the app's IOLoop in its thread
+        IOLoop.current().start()
```

Comment:
Because IPython no longer needs to run the zmq ioloop implementation, this thread can be done away with in most cases, and IPython could just share the main IOLoop. However! When event-loop integration is enabled in ipykernel (e.g. `%matplotlib qt`), the main event-loop becomes subordinate to the gui toolkit event-loop, and may not wake correctly for all dask events. So it may be safer to keep IPython in a thread anyway, despite not being *strictly* necessary.

Reply:
Isn't there something special about Qt needing to run in the main thread? It's been a while since I thought about this so could be misremembering or that could be outdated